docs/src/main/asciidoc/neo4j-guide.adoc

```diff
@@ -0,0 +1,545 @@
+The node is returned, a fruit entity created and than mapped to a `201` created response.
+
+A curl request against this path may look like this:
+
+[source, shell]
+----
+curl -v -X "POST" "http://localhost:8080/fruits" \
+     -H 'Content-Type: application/json; charset=utf-8' \
+     -d $'{
+  "name": "Banana"
+}'
+----
+
+The response contains an URI that shall return single nodes.
+
+=== Read single nodes
+
+This time, we signal a read only transaction.
+This way, a Neo4j cluster could select a follower.
+
+We also add some exception handling, in case the resource is called with an invalid id:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+@GET
+@Path("{id}")
+public CompletionStage<Response> getSingle(@PathParam("id") Long id) {
+    AsyncSession session = driver.asyncSession();
+    return session
+        .readTransactionAsync(tx ->
+            tx.runAsync("MATCH (f:Fruit) WHERE id(f) = $id RETURN f", Values.parameters("id", id))
+    )
+    .thenCompose(fn -> fn.singleAsync())
+    .handle((record, exception) -> {
+        if(exception != null) {
+            Throwable source = exception;
+            if(exception instanceof CompletionException) {
+                source = ((CompletionException)exception).getCause();
+            }
+            Status status = Status.INTERNAL_SERVER_ERROR;
+            if(source instanceof NoSuchRecordException) {
+                status = Status.NOT_FOUND;
+            }
+            return Response.status(status).build();
+        } else  {
+            return Response.ok(Fruit.from(record.get("f").asNode())).build();
+        }
+    })
+    .thenCompose(response -> session.closeAsync().thenApply(signal -> response));
+}
+----
+
+A request may look like this
+
+[source, shell]
+----
+curl localhost:8080/fruits/42
+----
+
+=== Deleting nodes
+
+Finally, we want to get rid of fruits again and we add the `DELETE` method:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+@DELETE
+@Path("{id}")
+public CompletionStage<Response> delete(@PathParam("id") Long id) {
+
+    AsyncSession session = driver.asyncSession();
+    return session
+        .writeTransactionAsync(tx ->
+            tx.runAsync("DELETE (f:Fruit) WHERE id(f) = $id", Values.parameters("id", id))
+        )
+        .thenCompose(fn -> fn.consumeAsync()) // <1>
+        .thenCompose(response -> session.closeAsync())
+        .thenApply(signal -> Response.noContent().build());
+}
+----
+<1> There is no result for us, only a summary of the query executed.
+
+A request may look like this
+
+[source, shell]
+----
+curl -X DELETE localhost:8080/fruits/42
+----
+
+And that's already the most simple CRUD application with one type of nodes.
+Feel free to add relationships to the model.
+One idea would be to model recipes that contain fruits.
+
+We add the super simple UI of another quick start guide.
```

Comment:
In the other guides, we usually skip the UI part as it's not very interesting in the Quarkus context.

See https://quarkus.io/guides/hibernate-search-guide#creating-a-frontend for instance.

I won't review this part for now. If you disagree, I'll review it again.

Reply:
It's basically a copy of the UI of the reactive Postgres guide. I think it's ok to leave it in the the actual getting started project, but I'll remove it from here.

Reply:
The problem is not with having an UI. The problem is that we really don't need to describe it and enter into the details.

See the example I pointed out: we just say "go grab the UI from GitHub" and that's OK for this type of guide where we don't really care about frontend development.

But sure, having a UI is nice.